src/student/repository/StudentRepository.ts

```diff
@@ -0,0 +1,6 @@
+import { Student } from 'src/student/model/Student';
+
+export interface StudentRepository {
+  save(student: Student): Promise<Student>;
+  findByEmail(email: string): Promise<Student>;
```

Comment:
Recuerda que se puede dar el caso de que esto retorne undefiend

```suggestion
  findByEmail(email: string): Promise<Student|undefined>;
```